simcon_project/users/views/student_management.py

```diff
@@ -0,0 +1,123 @@
+from users.forms import SendEmail, NewLabel
+from django.contrib.sites.shortcuts import get_current_site
+from django.shortcuts import render
+from django.utils.encoding import force_bytes
+from django.utils.http import urlsafe_base64_encode
+from users.models import Student, Researcher
+from django.core.mail import send_mail
+from users.models import SubjectLabel
+import django_tables2 as tables
+from django_tables2.config import RequestConfig
+from django.contrib import messages
+
+
+class StudentList(tables.Table):  #collects info from students to display on the table
+    first_name = tables.Column(verbose_name='First Name', accessor='students__first_name')
+    last_name = tables.Column(verbose_name='Last Name', accessor='students__last_name')
+    email = tables.Column(verbose_name='Email', accessor='students__email')
+    registered = tables.Column(verbose_name='Registered', accessor='students__registered')
+
+
+class LabelList(tables.Table):  #collects the table names
+    label_name = tables.Column(linkify={"viewname": "StudentManagement", "args": [tables.A("label_name")]},
+                               accessor='label_name', verbose_name='Label Name')
+
+
+def StudentManagement(request, name="All Students"):
+    # gets current researcher for use later
+    added_by = Researcher.objects.get(email=request.user)
```

Comment:
I'm getting an error at this line after I completed the registration form as a student and tried refreshing the student management page. Since I automatically got signed in as a student, I no longer had a researcher email. Though this would probably never come up in production, I think a solution would be to copy over David's `is_researcher` authentication test that he uses for his views